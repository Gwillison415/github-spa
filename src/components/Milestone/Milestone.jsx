import React from 'react';
import { List } from 'semantic-ui-react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import './Milestone.css';

const Milestone = ({ title, due, created }) => (
  <List.Item>
    <List.Content className="ui center aligned">
      <List.Header>{title}</List.Header>{created}
      <List.Description>{due}</List.Description>
    </List.Content>
  </List.Item>
);

const mapStateToProps = (state, { milestoneId }) => {
  console.log('state1', state);
  const milestone = state.milestones.milestonesByID[milestoneId];
  const { title, due, created } = milestone;
  return {
    title,
    due,
    created,
  };
};

Milestone.propTypes = {
  title: PropTypes.string.isRequired,
  due: PropTypes.string.isRequired,
  created: PropTypes.string.isRequired,
};

export default connect(
  mapStateToProps,
)(Milestone);
