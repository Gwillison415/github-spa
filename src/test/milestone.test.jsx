import React from 'react';
import toJson, { shallowToJson } from 'enzyme-to-json';
import { shallow, render, mount } from 'enzyme';
import {Milestone, mapStateToProps, mapDispatchToProps }from '../components/Milestone/Milestone';


const state = {
  milestones : {
    ids: [2601714],
    milestonesByID : {
    2601714 : {
      created: "2017-06-23T22:45:34Z",
      labels : ['sprint 1'],
      repoIssueNumber: 36,
      title : 'sprint 1',
      created: "2017-06-23T22:45:34Z",
      due:  "2018-06-23T22:45:34Z"
      }
    },
  },
  loadingMilestones: false
}

describe('Milestone entry component', () => {
  it('should render a component with props as specified ', () => {
    const component = shallow(
      <Milestone title={'sprint 1'}
      due={"2018-06-23T22:45:34Z"}
      milestoneId={2601714}
      created={"2017-06-23T22:45:34Z"}
      />,
    );
    expect(toJson(component)).toMatchSnapshot();
  });

  it('map\'s given State To Props ', () => {

    const expected = {
      title : 'sprint 1',
      created: "2017-06-23T22:45:34Z",
      due:  "2018-06-23T22:45:34Z",
    }
    expect(mapStateToProps(state, {milestoneId: 2601714})).toEqual(expected);
  })

  it('Should have a Cart Header in the center', () => {
    const component = shallow(
      <Milestone  />,
    );
    expect(component.find('.aligned').exists()).toBe(true);
  })
});
