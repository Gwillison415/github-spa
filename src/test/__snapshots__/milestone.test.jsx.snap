// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Milestone entry component should render a component with props as specified  1`] = `
<ListItem>
  <ListContent
    className="ui center aligned"
  >
    <ListHeader>
      sprint 1
    </ListHeader>
    06/23/2017 3:45 p.m.
    <ListDescription>
      2018-06-23T22:45:34Z
    </ListDescription>
  </ListContent>
</ListItem>
`;
